cogs/games.py

```diff
@@ -20,18 +20,203 @@
+                        if len(player_msg_list) > 3:
+                            player_msg_list = player_msg_list[-3:]
+                        timeout_dict[curr_msg.author] = time()
+                        txt_embed.set_field_at(
+                            0,
+                            name=f"hangman (category: {cat_name})",
+                            value=
+                            f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+                        )
+                        txt_embed.set_footer(text=last_line)
+                        await hg_msg.edit(embed=txt_embed)
+                        if num_mistakes == 6:
+                            await ctx.send(
+                                f"sorry everyone, {curr_msg.author} used your last chance, the right answer was `{word}`"
+                            )
+                    else:    # curr_guess not in not_guessed and not in word and in incorrect_guesses (else) => curr_guess is incorrect but already made
+                        timeout_dict[curr_msg.author] = time()
+                        player_msg_list.append(
+                            f"{curr_msg.author}, '{curr_guess}' was already guessed"
+                        )
+                        if len(player_msg_list) > 3:
+                            player_msg_list = player_msg_list[-3:]
+                        txt_embed.set_field_at(
+                            0,
+                            name=f"hangman (category: {cat_name})",
+                            value=
+                            f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+                        )
+                        await hg_msg.edit(embed=txt_embed)
+                elif curr_guess == word:
+                    not_guessed = set()
+                    first_line = "".join(char + " " for char in word).rstrip()
+                    player_msg_list.append(
+                        f"{curr_msg.author} guessed the entire word ('{word}') correctly!"
+                    )
+                    if len(player_msg_list) > 3:
+                        player_msg_list = player_msg_list[-3:]
+                    txt_embed.set_field_at(
+                        0,
+                        name=f"hangman (category: {cat_name})",
+                        value=
+                        f"`{first_line}`\n```{HANG_LIST[num_mistakes]}```\n{NEWLINE.join(player_msg_list)}"
+                    )
+                    await hg_msg.edit(embed=txt_embed)
+                    await ctx.send(
+                        f"congratulations {curr_msg.author}, you solved the hangman, but in a cool way"
+                    )
+                elif len(curr_guess) != 0:
+                    invalid_msg_count += 1
+                    player_msg_list.append(
+                        f"{curr_msg.author}, invalid guess"
```

Comment:
Don't write Invalid guess everytime, this is gonna be annoying when unrelated messages are sent (or sometimes people just, talk about the hangman too, and theyll get hit with a ton of "invalid guess")